monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ReferenceDataController.java

```diff
@@ -19,59 +19,383 @@
+    public SynonymCatalogModel getSynonymCatalog(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, final HttpServletRequest request) {
+        final SynonymCatalog synonymCatalog = getReferenceDataCatalog(tenant).getSynonymCatalog(name);
+        if (synonymCatalog == null) {
+            throw new NoSuchResourceException();
+        }
+
+        if (request.getMethod().equals(RequestMethod.HEAD.name())) {
+            return null;
+        }
+
+        final Collection<SynonymModel> entries =
+                synonymCatalog.openConnection(_contextFactory.getContext(tenant).getConfiguration()).getSynonyms()
+                        .stream().map(sc -> new SynonymModel(sc.getMasterTerm(),
+                        sc.getSynonyms().stream().filter(s -> !s.equals(sc.getMasterTerm()))
+                                .collect(Collectors.toList()))).collect(Collectors.toList());
+
+        return new SynonymCatalogModel(name, entries, synonymCatalog.isCaseSensitive());
+    }
+
+    @RolesAllowed(SecurityRoles.CONFIGURATION_EDITOR)
+    @RequestMapping(value = "/synonymCatalog/{name}", method = { RequestMethod.PUT },
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity putSynonymCatalog(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, @RequestBody final SynonymCatalogModel synonymCatalogModel) {
+        final TenantContext context = _contextFactory.getContext(tenant);
+        final FileResource resource = new FileResource(getResourceFile(context, name));
+        resource.write(stream -> {
+            for (final Synonym synonym : synonymCatalogModel.getEntries()) {
+                final List<String> line = new ArrayList<>(synonym.getSynonyms().size() + 1);
+                line.add(synonym.getMasterTerm());
+                line.addAll(synonym.getSynonyms());
+                stream.write((Joiner.on(",").join(line) + "\n").getBytes("UTF-8"));
+            }
+        });
+        final TextFileSynonymCatalog catalog =
+                new TextFileSynonymCatalog(name, resource.getFile(), synonymCatalogModel.isCaseSensitive(), "UTF-8");
+
+        _referenceDataDao.removeSynonymCatalog(context, name);
+        _referenceDataDao.addSynonymCatalog(context, catalog);
+        return ResponseEntity.created(ServletUriComponentsBuilder.fromCurrentRequestUri().build().toUri()).build();
+    }
+
+    @RolesAllowed(SecurityRoles.CONFIGURATION_EDITOR)
+    @RequestMapping(value = "/synonymCatalog/{name}", method = { RequestMethod.PUT },
+            consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity putSynonymCatalogFile(@PathVariable("tenant") final String tenant,
+            @PathVariable("name") final String name, @RequestParam("file") final MultipartFile sourceMultipartFile,
+            @RequestParam(value="casesensitive", defaultValue="true") final boolean isCaseSensitive,
```

Comment:
WhitespaceAround: '=' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)